js/classes/entities/midground/chad.js

```diff
@@ -242,25 +244,59 @@ class Chad {
         }
     };
 
+    knockback(direction, amount) {
+        // this.pos = Vector.add(this.pos, Vector.multiply(direction, amount));
+
+        this.knockbackForce = Vector.multiply(direction, amount * 100);
+        this.isKnockedBack = true;
+    }
+
+    manageKnockback() {
+        if (this.isKnockedBack) {
+            this.knockbackForce = Vector.multiply(this.knockbackForce, 0.5);
+            this.velocity = Vector.add(this.velocity, this.knockbackForce);
+
+            // If velocity is below a certain threshold, end knockback
+            if (Vector.magnitude(this.knockbackForce) < 0.1) {
+                this.isKnockedBack = false;
+            }
+        }
+    }
+
+
     /**
      * Increase the health of Chad by the provided amount
      * 
      * @param {number} amount the amount by which to increase Chad's health
      */
     restoreHealth(amount) {
```

Comment:
I like this. But will there be an issue with infinite health regen? He may become too powerful.